wsproto/connection.py

```diff
@@ -378,8 +378,10 @@ def _process_connection_request(self, event):
         if b'sec-websocket-version' not in headers:
             return ConnectionFailed(CloseReason.PROTOCOL_ERROR,
                                     "Missing Sec-WebSocket-Version header")
-        # XX FIXME: need to check Sec-Websocket-Version, and respond with a
-        # 400 if it's not what we expect
+        versions = _split_comma_header(headers[b'sec-websocket-version'])
```

Comment:
According to the spec, the client's `sec-websocket-version` header is not comma-delimited. It just has to be the literal string `b"13"`, nothing else. So instead of using `_split_comma_header`, this should just check that `headers[b'sec-websocket-version'] == b'13'`.

You can see this from the ABNF:
```
      Sec-WebSocket-Version-Client = version
      version = DIGIT | (NZDIGIT DIGIT) |
                ("1" DIGIT DIGIT) | ("2" DIGIT DIGIT)
                ; Limited to 0-255 range, with no leading zeros
```
And it's also in the text: "The request MUST include a header field with the name |Sec-WebSocket-Version|.  The value of this header field MUST be 13." (section 4.1). Also, "the |Sec-WebSocket-Version| header field MUST NOT appear more than once in an HTTP request." (section 11.3.5).

Reply:
Not sure which RFC you are reading, but I am using this section:
https://tools.ietf.org/html/rfc6455#section-4.4

which contains these examples:
```
 The response from the server might look as follows:

      HTTP/1.1 400 Bad Request
      ...
      Sec-WebSocket-Version: 13, 8, 7

   Note that the last response from the server might also look like:

      HTTP/1.1 400 Bad Request
      ...
      Sec-WebSocket-Version: 13
      Sec-WebSocket-Version: 8, 7
```

Reply:
Oh wait - you are talking about the client request of the handshake.
Sorry, I was referring to the server's response!

Reply:
Yeah, it's confusing -- they use `Sec-WebSocket-Version` for two almost, but not quite, entirely different things. The client always sends it, and it has to send exactly the string `b"13"` (for the final RFC 6455 version of the protocol). And, if the server doesn't like the version the client sent, then the server has to send back a comma-separated list of versions that it does like.

Reply:
Right - I will change this block to only check if the header value is `== b'13'`.

Reply:
Sorry, I totally forgot to push my changes - now this should be fixed.